.github/workflows/python.yml

```diff
@@ -150,6 +153,7 @@ jobs:
       ARROW_WITH_BROTLI: ON
       ARROW_BUILD_TESTS: OFF
       PYARROW_TEST_LARGE_MEMORY: ON
+      PYTEST_ARGS: -W error
```

Comment:
We could also leave it out here, and focus on getting no warnings on only linux?

Reply:
I would personally leave it out (I think enabling this for a smaller subset of builds will be easier to maintain)

Reply:
Sounds good! I'll update